internal/advertisementoperator/controller.go

```diff
@@ -309,7 +309,7 @@ func (r *AdvertisementReconciler) createVirtualKubelet(ctx context.Context, adv
 	klog.V(5).Infof("ClusterRoleBinding %s reconciled: %s", vkClusterRoleBinding.Name, op)
 
 	// Create the virtual Kubelet
-	vkDeployment, err := forge.CreateVkDeployment(adv, name, r.KubeletNamespace, r.VKImage, r.InitVKImage, nodeName, r.HomeClusterId)
+	vkDeployment, err := forge.VirtualKubeletDeployment(adv, "", name, r.KubeletNamespace, r.VKImage, r.InitVKImage, nodeName, r.HomeClusterId)
```

Comment:
After the refactoring, the new operator will only need to pass the clusterID value instead of the entire Advertisement (or ResourceOffer) resource. In the resourceOffer Operator, this method is called with nil Advertisement and a value in the remoteClusterID.

The double value is required for interoperability during the transition

Reply:
Got it, thanks